lib/Models/Terria.ts

```diff
@@ -1375,6 +1416,14 @@ export default class Terria {
       })
     );
 
+    const newItems: BaseModel[] = [];
+
+    await Promise.all(
+      newItemsRaw.map(async model => {
+        await this.pushAndLoadMapItems(model, newItems, errors);
+      })
+    );
+
     runInAction(() => (this.workbench.items = newItems));
```

Comment:
... continuing from previous comment

Here we could just have something like

```ts
    this.workbench.add(newItems);
```

This may need `workbench.add` to be refactored to return `Result` objects instead of throwing `TerriaError`s. Because at the moment, if one model fails to load in the workbench, then it all will fail.

Reply:
After reading the `workbench.add()` code, I feel that it requires too many code changes if moving the loading group logic there. The primary use case of this logic is to process init data.

Reply:
No problem, I'm happy to keep it in `Terria.ts` for the moment, but I think that in the long-term this should all be in `workbench.add` as it is directly related to adding models to the workbench, and doesn't really have anything to do with core Terria functionality - `Terria.ts` is too large at the moment.